pinot-broker/src/test/java/org/apache/pinot/broker/routing/HelixExternalViewBasedTimeBoundaryServiceTest.java

```diff
@@ -0,0 +1,164 @@
+import org.apache.commons.lang.StringUtils;
+import org.apache.helix.ZNRecord;
+import org.apache.helix.manager.zk.ZNRecordSerializer;
+import org.apache.helix.manager.zk.ZkBaseDataAccessor;
+import org.apache.helix.manager.zk.ZkClient;
+import org.apache.helix.model.ExternalView;
+import org.apache.helix.store.zk.ZkHelixPropertyStore;
+import org.apache.pinot.broker.routing.TimeBoundaryService.TimeBoundaryInfo;
+import org.apache.pinot.common.config.TableConfig;
+import org.apache.pinot.common.config.TableNameBuilder;
+import org.apache.pinot.common.data.FieldSpec;
+import org.apache.pinot.common.data.Schema;
+import org.apache.pinot.common.metadata.ZKMetadataProvider;
+import org.apache.pinot.common.metadata.segment.OfflineSegmentZKMetadata;
+import org.apache.pinot.common.utils.CommonConstants;
+import org.apache.pinot.common.utils.StringUtil;
+import org.apache.pinot.common.utils.ZkStarter;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertNull;
+
+
+public class HelixExternalViewBasedTimeBoundaryServiceTest {
+  private static final String TIME_COLUMN = "time";
+
+  private ZkStarter.ZookeeperInstance _zookeeperInstance;
+  private ZkClient _zkClient;
+  private ZkHelixPropertyStore<ZNRecord> _propertyStore;
+
+  @BeforeClass
+  public void setUp() {
+    _zookeeperInstance = ZkStarter.startLocalZkServer();
+
+    _zkClient = new ZkClient(StringUtil.join("/", StringUtils.chomp(ZkStarter.DEFAULT_ZK_STR, "/")),
+        ZkClient.DEFAULT_SESSION_TIMEOUT, ZkClient.DEFAULT_CONNECTION_TIMEOUT, new ZNRecordSerializer());
+    String helixClusterName = "TestTimeBoundaryService";
+    _zkClient.deleteRecursively("/" + helixClusterName + "/PROPERTYSTORE");
+    _zkClient.createPersistent("/" + helixClusterName + "/PROPERTYSTORE", true);
+    _propertyStore = new ZkHelixPropertyStore<>(new ZkBaseDataAccessor<ZNRecord>(_zkClient),
+        "/" + helixClusterName + "/PROPERTYSTORE", null);
+  }
+
+  @AfterClass
+  public void tearDown() {
+    _zkClient.close();
+    ZkStarter.stopLocalZkServer(_zookeeperInstance);
+  }
+
+  @Test
+  public void testExternalViewBasedTimeBoundaryService()
+      throws Exception {
+    HelixExternalViewBasedTimeBoundaryService timeBoundaryService =
+        new HelixExternalViewBasedTimeBoundaryService(_propertyStore);
+
+    int tableIndex = 0;
+    for (TimeUnit timeUnit : TimeUnit.values()) {
+      String rawTableName = "table" + tableIndex;
+      String offlineTableName = TableNameBuilder.OFFLINE.tableNameWithType(rawTableName);
+      String realtimeTableName = TableNameBuilder.REALTIME.tableNameWithType(rawTableName);
+      addTableConfig(rawTableName, timeUnit, "daily");
+      addSchema(rawTableName, timeUnit);
+      addSegmentZKMetadata(rawTableName, 10 + tableIndex, timeUnit);
```

Comment:
If you can please add a final int 10 and final int 9 that would also explain where these came from. It will help if the test fails for any reason. It will also help if we want to modify the test. thanks,

Reply:
Done